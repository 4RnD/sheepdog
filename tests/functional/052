#!/bin/bash

# Test force cluster recovery with new nodes

. ./common

for i in 0 1 2 3; do
    _start_sheep $i
done
_wait_for_sheep 4
_cluster_format
$COLLIE vdi create test 20M -P
$COLLIE cluster info | _filter_cluster_info

$COLLIE cluster shutdown
_wait_for_sheep_stop

for i in 0 1 2 4; do
    _start_sheep $i
done
_wait_for_sheep 4
for i in 0 1 2 4; do
    $COLLIE cluster info -p 700$i | _filter_cluster_info
    $COLLIE vdi list | _filter_short_date
done

echo yes | $COLLIE cluster recover force
echo ""

_wait_for_sheep_recovery 0
$COLLIE vdi check test | sort

for i in 0 1 2 4; do
    $COLLIE cluster info -p 700$i | _filter_cluster_info
    $COLLIE vdi list | _filter_short_date
done

_start_sheep 3
_wait_for_sheep 5
for i in 0 1 2 3 4; do
    $COLLIE cluster info -p 700$i | _filter_cluster_info
    $COLLIE vdi list | _filter_short_date
done
