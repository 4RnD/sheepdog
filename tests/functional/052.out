QA output created by 052
using backend plain store
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Cluster status: Waiting for other nodes to join cluster

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Cluster status: Waiting for other nodes to join cluster

Failed to read object 807c2b2500000000 Waiting for other nodes to join cluster
Failed to read inode header
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
Caution! Please try starting all the cluster nodes normally before
running this command.

The cluster may need to be force recovered if:
  - the master node fails to start because of epoch mismatch; or
  - some nodes fail to start after a cluster shutdown.

Are you sure you want to continue? [yes/no]: 
finish check&repair test
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      3 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128, 127.0.0.1:7006:128]
DATE      2 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7006:128]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128, 127.0.0.1:7003:128, 127.0.0.1:7004:128, 127.0.0.1:7005:128]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   20 MB   20 MB  0.0 MB DATE   7c2b25      6                22
